internal/clients/team4/iito.go

```diff
@@ -0,0 +1,174 @@
+		c.internalParam.fairness,
+		c.internalParam.collaboration,
+	})
+	// selflessLevel is the opposite of greedinessLevel in GetGiftRequests function.
+	selflessLevel := mat.Dot(importance, parameters)
+
+	// Initialising thresholds used.
+	resourcesThreshold := 100.0
+
+	// have a cap at 100 in case we are too selfless
+	wealthGoal := math.Max(selflessLevel * float64(ourResources), resourcesThreshold)
+	
+	// giftSize is the maximum we can afford giving away.
+	giftSize :=  float64(ourResources) - wealthGoal
+
+	// Lets just return the min of "giftSize" and giftOffer.
+	// The idea is that we are very careful in the function that 
+	// makes offers, so here we just keep our word as long as it does not
+	// put us in a stressful state / our selfishness and greediness have 
+	// increased since we made the offer.
+	actualGift := math.Min(float64(giftOffer), giftSize)
+	return shared.Resources(actualGift)
+}
+
+// SentGift is executed at the end of each turn and notifies clients that
+// their gift was successfully sent, along with the offer details.
+// COMPULSORY, you need to implement this method
+func (c *client) SentGift(sent shared.Resources, to shared.ClientID) {
+	// You can check your updated resources like this:
+	myResources := c.ServerReadHandle.GetGameState().ClientInfo.Resources
+	percentageDonated := float64(sent / (sent + myResources))
+	bumpThreshold := 0.2
+	currentGreediness := c.internalParam.greediness
+	maxGreedBump := math.Min(currentGreediness + bumpThreshold, 1.0)
+	
+	currentSelfishness := c.internalParam.selfishness
+	maxSelfishBump := math.Min(currentSelfishness + bumpThreshold, 1.0)
+	
+	// Bump up greediness and selfishness.
+	// We use a maximum bump to cap at 1 and reduce the volatility.
+	c.internalParam.greediness = math.Min(currentGreediness + percentageDonated, maxGreedBump)
+	c.internalParam.selfishness = math.Min(currentSelfishness + percentageDonated, maxSelfishBump)
+	
+
+}
+
+// ReceivedGift is executed at the end of each turn and notifies clients that
+// their gift was successfully received, along with the offer details.
+// COMPULSORY, you need to implement this method
+func (c *client) ReceivedGift(received shared.Resources, from shared.ClientID) {
+	// You can check your updated resources like this:
+	myResources := c.ServerReadHandle.GetGameState().ClientInfo.Resources
+	percentageDonated := float64(received) / float64(myResources)
+	bumpThreshold := 0.2
+
+	currentCollab := c.internalParam.collaboration
+	maxCollabBump := math.Min(currentCollab + bumpThreshold, 1.0)
+
+	currentAgentTrust := c.trustMatrix.GetClientTrust(from)
+	maxTrustBump := math.Min(currentAgentTrust + bumpThreshold, 1.0)
+	// Bump up collaboration and agent trust.
+	// We use a maximum bump to cap at 1 and reduce the volatility.
+	c.internalParam.collaboration = math.Min(currentCollab + percentageDonated, maxCollabBump)
+	c.trustMatrix.SetClientTrust(from, math.Min(currentAgentTrust + percentageDonated, maxTrustBump))
```

Comment:
this discards our previously built trust if maxTrustBump is reached. Is that expected?

Reply:
This is resolved based on our chat.